tf-build-tools/tf-maven-plugin/src/main/java/com/deliveredtechnologies/maven/terraform/TerraformPackage.java

```diff
@@ -149,6 +134,79 @@ public boolean accept(File pathname) {
     }
   }
 
+  private void copyTfRootDir(Path sourceDir, Path targetDir) throws IOException {
+    List<Path> files = Files.walk(sourceDir, 1)
+        .filter(path -> !path.equals(sourceDir))
+        .collect(Collectors.toList());
+
+    for (Path file : files.stream().filter(f -> !excludedFiles.contains(f.getFileName().toString())).collect(Collectors.toList())) {
+      if (file.toFile().isDirectory()) {
+        FileUtils.copyDirectory(file.toFile(), targetDir.resolve(file.getFileName().toString()).toFile(), new FileFilter() {
+          @Override
+          public boolean accept(File pathname) {
+            return excludedFiles.stream().noneMatch(excludedFile -> pathname.getName().endsWith(excludedFile));
+          }
+        });
+      } else {
+        Files.copy(file, targetDir.resolve(file.getFileName().toString()));
+      }
+    }
+  }
+
+  private void copyTfVarFiles(Path sourceDir, String files, Path targetDir) throws IOException {
+    List<Path> filePaths = StringUtils.isEmpty(files)
+        ? new ArrayList<>()
+        : Arrays.stream(files.split(","))
+        .map(file -> sourceDir.resolve(file.trim()))
+        .collect(Collectors.toList());
+    logger.debug(String.format("tfVarFiles is %1$s", filePaths));
+
+    for (Path filePath : filePaths) {
+      Files.copy(filePath, targetDir.resolve(filePath.getFileName().toString().replaceAll(".tfvars", "").concat(".auto.tfvars")));
+    }
+  }
+
+  private void saveTfVarsAsTfVarFile(Path sourceDir, Object tfVars, Path targetDir, String targetFileName) throws IOException {
+    if (tfVars == null) {
+      return;
+    }
+
+    String fileContent;
+    Map<String,Object> tfVarsMap;
+    ObjectMapper objectMapper = new ObjectMapper();
+    File sourceFile = new File(sourceDir.resolve(targetFileName).toUri());
+    File targetFile = new File(targetDir.resolve(targetFileName).toUri());
+
+    logger.debug(String.format("tfVars is %1$s", tfVars.toString()));
+
+    if (sourceFile.exists()) {
+      tfVarsMap = objectMapper.readValue(new FileInputStream(sourceFile), Map.class);
```

Comment:
I guess this is incase there is already a file with this name? and the assumption is that new values override. What does that mean for values that are in the existing file that are no longer relevant but wont be overriden